src/listpack.c

```diff
@@ -551,31 +557,37 @@ unsigned char *lpGet(unsigned char *p, int64_t *count, unsigned char *intbuf) {
         negstart = UINT64_MAX; /* 7 bit ints are always positive. */
         negmax = 0;
         uval = p[0] & 0x7f;
+        if (size) *size = LP_ENCODING_7BIT_UINT_ENTRY_SIZE;
     } else if (LP_ENCODING_IS_6BIT_STR(p[0])) {
         *count = LP_ENCODING_6BIT_STR_LEN(p);
+        if (size) *size = 2 + *count;
```

Comment:
1) It should be `if (size) *size = 1 + *count + lpEncodeBacklen(NULL, *count + 1);`, It brings no significant improvement.
2) The `backlen` of 12bit and 32bit string are dynamic lengths, so we can't assume their entry length.
3) There is no way for listpack to store a small string in a large encoding type, even if `replace` operation, it will become `6BIT_STR` when replacing `32BIT_STRING` to a small string.

Reply:
i meant for a case that someone produces a specially crafted RESTORE payload, then stores a small item in a large encoding type.
then maybe there's a mismatch between the encoded len and the actual prevlen bytes.
i.e. the listpack can still be iterated correctly from head to tail and tail to head, and the integrity validation passes.
but some piece of code that assumes it knows the prevlen bytes fails...
but i guess even for forward iteration, must know the size of the prevlen bytes.. that's part of the design of the listpack...